frontends/web/src/components/forms/input-number.tsx

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright 2024 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ChangeEvent, useCallback } from 'react';
+import { Input, TInputProps } from './input';
+
+type Props = Omit<TInputProps, 'ref' | 'onInput'> & {
+  onChange?: (event: ChangeEvent<HTMLInputElement>) => void
```

Comment:
I was not sure if this should be mandatory, as the whole logic onPaste calls onChange and does not much else 🤔